bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/handler/ZonePlayerHandler.java

```diff
@@ -1673,14 +1755,8 @@ public void setVolume(Command command) {
             } else {
                 return;
             }
-            inputs.put("Channel", "Master");
-            inputs.put("DesiredVolume", newValue);
-
-            Map<String, String> result = service.invokeAction(this, "RenderingControl", "SetVolume", inputs);
-
-            for (String variable : result.keySet()) {
-                this.onValueReceived(variable, result.get(variable), "RenderingControl");
-            }
+            executeAction(SERVICE_RENDERING_CONTROL, ACTION_SET_VOLUME,
+                    Map.of("InstanceID", "0", "Channel", "Master", "DesiredVolume", newValue));
```

Comment:
Where did `"InstanceID"` come from? Is this a fix for something?

Reply:
Yes. In fact InstanceID is missing in several action calls. I suppose it is optional. I will undo this adding.

Reply:
InstanceID parameter removed.